packetbeat/protos/dns/dns.go

```diff
@@ -275,35 +280,59 @@ func (dns *Dns) ConnectionTimeout() time.Duration {
 }
 
 func (dns *Dns) receivedDnsRequest(tuple *DnsTuple, msg *DnsMessage) {
-	logp.Debug("dns", "Processing query. %s", tuple.String())
+	debug("Processing query. %s", tuple.String())
 
 	trans := dns.deleteTransaction(tuple.Hashable())
 	if trans != nil {
 		// This happens if a client puts multiple requests in flight
 		// with the same ID.
 		trans.Notes = append(trans.Notes, DuplicateQueryMsg.Error())
-		logp.Debug("dns", "%s %s", DuplicateQueryMsg.Error(), tuple.String())
+		debug("%s %s", DuplicateQueryMsg.Error(), tuple.String())
 		dns.publishTransaction(trans)
 		dns.deleteTransaction(trans.tuple.Hashable())
 	}
 
 	trans = newTransaction(msg.Ts, *tuple, *msg.CmdlineTuple)
+
+	if tuple.Transport == TransportUdp && !msg.Edns && msg.Length > MaxDnsPacketSize {
+		trans.Notes = append(trans.Notes, UdpPacketTooLarge.Error())
+		debug("%s", UdpPacketTooLarge.Error())
+	}
+
 	dns.transactions.Put(tuple.Hashable(), trans)
 	trans.Request = msg
 }
 
 func (dns *Dns) receivedDnsResponse(tuple *DnsTuple, msg *DnsMessage) {
-	logp.Debug("dns", "Processing response. %s", tuple.String())
+	debug("Processing response. %s", tuple.String())
 
 	trans := dns.getTransaction(tuple.RevHashable())
 	if trans == nil {
 		trans = newTransaction(msg.Ts, tuple.Reverse(), common.CmdlineTuple{
 			Src: msg.CmdlineTuple.Dst, Dst: msg.CmdlineTuple.Src})
 		trans.Notes = append(trans.Notes, OrphanedResponse.Error())
-		logp.Debug("dns", "%s %s", OrphanedResponse.Error(), tuple.String())
+		debug("%s %s", OrphanedResponse.Error(), tuple.String())
 	}
 
 	trans.Response = msg
+
+	if tuple.Transport == TransportUdp && !msg.Edns && msg.Length > MaxDnsPacketSize {
+		trans.Notes = append(trans.Notes, UdpPacketTooLarge.ResponseError())
+		debug("%s", UdpPacketTooLarge.ResponseError())
+	}
+
+	request := trans.Request
+	if request != nil {
+		switch {
+		case request.Edns && !msg.Edns:
+			trans.Notes = append(trans.Notes, RespEdnsNoSupport.Error())
+			debug("%s %s", RespEdnsNoSupport.Error(), tuple.String())
+		case !request.Edns && msg.Edns:
+			trans.Notes = append(trans.Notes, RespEdnsUnexpected.Error())
+			debug("%s %s", RespEdnsUnexpected.Error(), tuple.String())
+		}
```

Comment:
I think that the first case is likely. It shows that the dns server is either mis-configured or simply does not support EDNS.
The second case is unlikely.